src/App.js

```diff
@@ -14,11 +15,114 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    goods: goodsFromServer,
+    sortedBy: '',
+    isReversed: false,
+    isVisible: false,
+  }
+
+  showList = () => {
+    this.setState({ isVisible: true });
+  }
+
+  hideList = () => {
+    this.setState({ isVisible: false });
+  }
+
+  nameSortList = () => {
+    this.setState({ sortedBy: 'name' });
+  }
+
+  lengthSortList = () => {
+    this.setState({ sortedBy: 'length' });
+  }
+
+  reverseList = () => {
+    this.setState(state => ({
+      isReversed: !state.isReversed,
+    }));
+  }
+
+  resetList = () => {
+    this.setState({
+      sortedBy: '',
+      isReversed: false,
+    });
+  }
+
+  render() {
+    const { goods, isVisible, sortedBy, isReversed } = this.state;
+    const copyGoods = goods.slice();
+
+    return (
+      <div className="App">
+        <h1>Goods</h1>
+
+        {!isVisible
+        && (
+          <button
+            type="button"
+            className="startbtn"
+            onClick={this.showList}
+          >
+            start
+          </button>
+        )
+        }
+        {isVisible
+          && (
+            <>
+              <GoodsList
+                goods={copyGoods}
+                sortedBy={sortedBy}
+                isReversed={isReversed}
```

Comment:
just pass already prepared array, no need to do such things inside GoodsList component

Reply:
remove props that you don't use